samples/net/sntp/src/main.c

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright (c) 2024 Embeint Inc
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+ #include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(net_sntp_sample, LOG_LEVEL_DBG);
+
+#include <time.h>
+
+#include <zephyr/net/socket.h>
+#include <zephyr/net/socket_service.h>
+#include <zephyr/net/sntp.h>
+
+#include "net_sample_common.h"
+
+static K_SEM_DEFINE(sntp_async_received, 0, 1);
+static void sntp_service_handler(struct net_socket_service_event *pev);
+
+NET_SOCKET_SERVICE_SYNC_DEFINE_STATIC(service_sntp_async, sntp_service_handler, 1);
+
+int dns_query(const char *host, uint16_t port, int family, int socktype, struct sockaddr *addr,
+	      socklen_t *addrlen)
+{
+	struct zsock_addrinfo hints = {
```

Comment:
For the networking samples, we have tried to mainly use the BSD socket API (so calling without `zsock_` which means that `CONFIG_POSIX_API` needs to be enabled).

Reply:
The downside for this sample is that `CONFIG_POSIX_API` looks to break the `time` function, meaning you can no longer compare the SNTP time to the local system time if run on `native_posix`.

Reply:
Ok, fair enough, there is certainly things in Posix that do not work properly yet.